src/predict_request_validation_utils.cpp

```diff
@@ -653,9 +653,12 @@ Status RequestValidator<RequestType, InputTensorType, IteratorType, ShapeType>::
             if (!status.ok())
                 return status;
 
-            status = checkBinaryBatchSizeMismatch(proto, batchSize, finalStatus, batchingMode, shapeMode);
-            if (!status.ok())
-                return status;
+            const KFSRequest* r = dynamic_cast<const KFSRequest*>(&request);
```

Comment:
Template shouldn't care about types with with it will be used. This should be pushed lower into checkBinarySizeMismatch. Or we should define template for checking if we want ot check for BinaryBatchSizeMismatch with default specialization always returning true, and specialization for KFS that will decide by information from request.

Reply:
I think the 2nd solution looks better.